### PR TITLE
Copy auth to child anomalies

### PR DESCRIPTION
**Pull request by jacksonargo**
Set auth for child anomalies.

Fixup from https://github.com/startreedata/thirdeye/pull/898

### CODE REVIEW

File: thirdeye-persistence/src/main/java/ai/startree/thirdeye/datalayer/bao/AnomalyManagerImpl.java [hunk 1]
```diff
@@ -173,6 +173,7 @@ private Set<Long> saveChildAnomalies(final AnomalyDTO parentAnomaly,
           throw new IllegalArgumentException("Loop detected! Child anomaly referencing ancestor");
         }
       }
+      child.setAuth(parentAnomaly.getAuth());
```
Comment (cyrilou242) [inline thirdeye-persistence/src/main/java/ai/startree/thirdeye/datalayer/bao/AnomalyManagerImpl.java:176]: very clean 👍

File: thirdeye-worker/src/main/java/ai/startree/thirdeye/worker/task/runner/DetectionPipelineTaskRunner.java [hunk 2]
```diff
@@ -186,25 +186,11 @@ public OperatorResult run(final AlertDTO alert, final Interval detectionInterval
...
 
   private void postExecution(final OperatorResult result) {
-    final List<AnomalyDTO> anomalies = result.getAnomalies();
-    if (anomalies == null) {
-      return;
-    }
-    anomalies.stream().map(this::setAnomalyAuth).forEach(anomalyDTO -> {
-      final Long id = anomalyDao.save(anomalyDTO);
-      if (id == null) {
-        LOG.error("Failed to store anomaly: {}", anomalyDTO);
```
[the remark this answers is not in the document]
Comment (cyrilou242) [reply thirdeye-worker/src/main/java/ai/startree/thirdeye/worker/task/runner/DetectionPipelineTaskRunner.java:196]: or maybe in the AnomalyManager this way we keep DatabaseService without logs.

[the remark this answers is not in the document]
Comment (jacksonargo) [reply thirdeye-worker/src/main/java/ai/startree/thirdeye/worker/task/runner/DetectionPipelineTaskRunner.java:196]: 🙆‍♂️

**Review by cyrilou242 — APPROVED** [review]
LGTM! great stuff.

**Pull request merged**